### ISSUE TITLE
How to capture and return value?

### ISSUE THREAD

**Issue by bulldetektor** [NagRock/ts-mockito#16]
I'm trying to do both a capture and return on the same mock, but I can't figure out how to set it up. This is my attempt so far:

```
const mockedDS = mock(MyDataService);
const whenCall = when(mockedDS.update(anything(), anything());
whenCall.thenReturn("OK");
const firstCaptor = new Captor<Array<Employee>>();
const secondCaptor = new Captor<TimePeriod>();
whenCall.thenCapture(firstCaptor, secondCaptor);
```

It seems that I can only do one or the other? Depending on whether I call `thenCapture` before `thenReturn` or vice versa it will only capture OR return the value.

Btw. What I'd really like to write is this;
```
const mockedDS = mock(MyDataService);
const firstCaptor = new Captor<Array<Employee>>();
const secondCaptor = new Captor<TimePeriod>();
when(mockedDS.update(anything(), anything())
    .thenCapture(firstCaptor, secondCaptor);
    .thenReturn("OK");
```
...but both `thenReturn` and `thenCapture` returns `void`

**Comment by NagRock** [comment]
Thanks for reporting, I will add this functionality soon. Currently you can use `thenCall` to capture and return value as workaround.

**Comment by bulldetektor** [comment]
Yes, that's what I ended up doing. Works fine, just not as fluent as I'd wish it was. But apart from that; ts-mockito is just lovely to work with :)

**Comment by NagRock** [comment]
Thanks for appreciation. As I wrote earlier I will add functionality to capture and stub mock with API that you tried to use: `when(mock.smthn(anything()))
    .thenCapture(firstCaptor);
    .thenReturn("something");`

**Comment by NagRock** [comment]
Should be fixed in [2.0 version](https://github.com/NagRock/ts-mockito/releases/tag/v2.0.0). Please reopen if something is still wrong.

**Comment by alexleen** [comment]
> Should be fixed in 2.0 version. Please reopen if something is still wrong.

What was the end result? I'm using 2.6.1 and can't seem to find a way to do this. What I would like is something similar to your comment above: `when(mock.smthn(anything())).thenCapture(firstCaptor).thenReturn(firstCaptor);`

**Comment by matitalatina** [comment]
@alexleen I had your issue and I ended up using https://github.com/NagRock/ts-mockito/#custom-function
Just create a function that returns the first argument.
